### PR TITLE
Fix/no default cookies

### PR DESCRIPTION
**Pull request by juanig1**
When no default cookie is present, adding cookies for a particular scenario does not work

Closes [1035](https://github.com/artilleryio/artillery/issues/1035), [1420](https://github.com/artilleryio/artillery/issues/1420)

### CODE REVIEW

**Comment by hassy** [comment]
LGTM! is this good to go @juanig1 or are you still working on it?

**Comment by juanig1** [comment]
Found this [commit message](https://github.com/artilleryio/artillery/commit/1ef06ba9e322f96fbf0387243fb4f4ffb2a93104) warning about CPU usage.
Do you remember _how_ bad was it? Maybe adding a flag to disable the cookie jar may be a nice addition to this PR

**Comment by hassy** [comment]
Ah, good find on that PR!

It looks like the real issue then is that when `set-cookie` header is present, the cookie jar is still not enabled. We should figure out why instead of enabling the cookie jar by default.

File: core/lib/engine_http.js [hunk 1]
```diff
@@ -398,13 +398,14 @@ HttpEngine.prototype.step = function step(requestSpec, ee, opts) {
         requestParams.headers = templatedHeaders;
 
         if (
-          typeof params.cookie === 'object' ||
+          typeof requestParams.cookie === 'object' ||
```
Comment (juanig1) [inline core/lib/engine_http.js:401]: requestparam includes the changes made by the processor

**Review by yaswanthpflegia — APPROVED** [review]

**Pull request merged**